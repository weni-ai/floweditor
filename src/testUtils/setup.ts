/* eslint-disable no-redeclare */
import { Console } from 'console';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Exit } from 'flowTypes';
import { RenderNode } from 'store/flowContext';

// Declare custom matcher types
declare global {
  namespace jest {
    interface Matchers<R> {
      toPointTo(renderNode: RenderNode): R;
      toHaveExitThatPointsTo(renderNode: RenderNode): R;
      toHaveInboundFrom(exit: Exit): R;
      toHaveExitWithDestination(): R;
      toHaveInboundConnections(): R;
      toHavePayload(action: string, payload: any): R;
      toHaveReduxActions(actions: string[]): R;
      toMatchCallSnapshot(snapshotName?: string): R;
    }
  }
}

// Ensure console logs are visible while running tests https://github.com/facebook/jest/issues/3853
global.console = new Console(process.stderr, process.stderr);

// Configure Enzyme adapter
configure({ adapter: new Adapter() });

// RAF shim
// tslint:disable-next-line:ban-types
(global as any).requestAnimationFrame = (callback: Function) => {
  setTimeout(callback, 0);
};
