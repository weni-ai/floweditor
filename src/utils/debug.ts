import { FlowStoreProps } from 'components/flow/Flow';
import { DebugState } from 'store/editor';

import mutate from 'immutability-helper';

/* istanbul ignore next -- @preserve */
export default class Debug {
  private props: FlowStoreProps;
  private state: DebugState;

  constructor(props: FlowStoreProps, initial: DebugState) {
    this.props = props;
    this.state = initial || { showUUIDs: false };
  }

  public showUUIDs(): DebugState {
    const updated = mutate(this.state, { $merge: { showUUIDs: true } });
    this.props.mergeEditorState({ debug: updated });
    return updated;
  }
}
