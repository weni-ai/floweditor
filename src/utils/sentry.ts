import * as Sentry from '@sentry/react';

const initSentry = (dsn: string) => {
  Sentry.init({
    dsn,
    integrations: [
      Sentry.browserTracingIntegration(),
      Sentry.httpClientIntegration({
        failedRequestStatusCodes: [[400, 499], [500, 599]],
      }),
      Sentry.replayIntegration(),
      Sentry.replayCanvasIntegration(),
    ],
    tracesSampleRate: 1.0,
    replaysSessionSampleRate: 0.1,
    replaysOnErrorSampleRate: 1.0,
  });
};

export default initSentry;
