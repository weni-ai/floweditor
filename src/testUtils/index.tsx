// TODO: Remove use of Function
// tslint:disable:ban-types
import { ConfigProviderContext, fakePropType } from 'config/ConfigProvider';
import { mount, ReactWrapper, shallow, ShallowWrapper } from 'enzyme';
import { FlowDefinition, FlowEditorConfig } from 'flowTypes';
import mutate, { Query } from 'immutability-helper';
import * as React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import createStore from 'store/createStore';
import { RenderNodeMap } from 'store/flowContext';
import { getFlowComponents } from 'store/helpers';
import AppState, { initialState } from 'store/state';
import config from 'test/config';
import * as matchers from 'testUtils/matchers';
import { merge, set } from 'utils';

// we need to use require syntax to bust implicit any
const boring: FlowDefinition = require('test/flows/boring.json');

// force our matchers to be read in
const match = matchers;

export interface Resp {
  assets: { [key: string]: any }[];
}

export interface QueryString {
  [key: string]: string;
}

export const contextTypes: { [key: string]: Function } = {
  config: fakePropType,
  store: fakePropType,
  assetService: fakePropType,
};

export const baseState: AppState = mutate(initialState, {
  flowContext: merge({
    definition: require('test/flows/colors.json') as FlowDefinition,
  }),
  editorState: merge({
    language: { id: 'eng', name: 'English' },
  }),
});

const flowEditorConfig: FlowEditorConfig = config;

export const configProviderContext: ConfigProviderContext = {
  config: flowEditorConfig,
};

export const setMock = (
  implementation?: (...args: any[]) => any,
): Query<jest.Mock> => set(jest.fn(implementation));

/**
 * Compose setup method for component tests
 */
export const composeSetup = <P extends {}>(
  Component: React.ComponentClass | React.SFC,
  baseProps: P = {} as any,
  baseDuxState: AppState | Partial<AppState> = baseState,
  baseContext: ConfigProviderContext = configProviderContext,
) => (
  shallowRender: boolean = true,
  propOverrides: Query<P | Partial<P>> = {},
  duxStateOverrides: Query<AppState | Partial<AppState>> = {},
  contextOverrides: Query<
    ConfigProviderContext | Partial<ConfigProviderContext>
  > = {},
  childContextTypeOverrides: { [key: string]: Function } = {},
  children: (JSX.Element | React.ComponentClass | React.SFC)[] = [],
) => {
  const props = mutate(baseProps, propOverrides);
  const store = createStore(mutate(
    baseDuxState,
    duxStateOverrides,
  ) as AppState);
  let context = mutate(baseContext, merge({ store }));

  if (Object.keys(duxStateOverrides).length > 0) {
    context = mutate(context, duxStateOverrides);
  }

  if (Object.keys(contextOverrides).length > 0) {
    context = mutate(context, contextOverrides as ConfigProviderContext);
  }

  const childContextTypes: { [contextProp: string]: Function } = mutate(
    contextTypes,
    childContextTypeOverrides,
  );
  // tslint:disable-next-line:ban-types
  const wrapper = (shallowRender ? (shallow as Function) : (mount as Function))(
    <Component {...props}>{children}</Component>,
    {
      context,
      childContextTypes,
    },
  );

  return {
    wrapper,
    props,
    context,
    instance: wrapper.instance(),
  };
};

export const composeSpy = (obj: Record<string, any> | React.ComponentClass) => (
  instanceMethod: string,
) =>
  jest.spyOn(
    (obj as React.ComponentClass).prototype || obj,
    instanceMethod as any,
  );

/**
 * Wait for promises in queue to resolve
 */
export const flushPromises = () =>
  new Promise((resolve: any) => setImmediate(resolve));

/**
 * Restore spy mocks (distinct from mocks created w/ jest.fn()).
 * Use to declaratively restore multiple spy mocks,
 * otherwise just call .mockRestore() on lone spy.
 */
export const restoreSpies = (...spies: jest.SpyInstance<any>[]) =>
  spies.forEach((spy: any) => spy.mockRestore());

// To-do: type this method's output, can pass it prop generic ingested by getComponentTestUtils
/**
 * NOTE: borrowed from EventBrite: https://github.com/eventbrite/javascript/blob/master/react/testing.md#finding-nodes
 * Finds all instances of components in the rendered `componentWrapper` that are DOM components
 * with the `data-spec` attribute matching `name`.
 * @param {ReactWrapper|ShallowWrapper} componentWrapper - Rendered componentWrapper (result of mount, shallow, or render)
 * @param  {string} snacName - Name of `data-spec` attribute value to find
 * @param {string|Function} (Optional) typeFilter - (Optional) Expected type of the wrappers (defaults to all HTML tags)
 * @returns {ReactWrapper|ReactWrapper[]|ShallowWrapper|ShallowWrapper[]} Matching DOM components
 */
export const getSpecWrapper = (
  componentWrapper: ReactWrapper<{}, {}> | ShallowWrapper<{}, {}>,
  specName: string,
  attributeName: string = 'data-spec',
): any => componentWrapper.find(`[${attributeName}="${specName}"]`);

export const composeDuxState = (
  query: Query<AppState | Partial<AppState>> = {},
  duxState = baseState,
) => mutate(duxState, query);

export const composeComponentTestUtils = <P extends {}>(
  Component: React.ComponentClass | React.SFC | any,
  baseProps: P = {} as any,
  baseDuxState: AppState | Partial<AppState> = baseState,
  baseContext: ConfigProviderContext = configProviderContext,
) => ({
  setup: composeSetup<P>(Component, baseProps, baseDuxState, baseContext),
  spyOn: composeSpy(Component),
});

export const prepMockDuxState = (): {
  testNodes: RenderNodeMap;
  mockDuxState: AppState;
} => {
  const testNodes = getFlowComponents(boring).renderNodeMap;

  return {
    testNodes,
    mockDuxState: {
      ...initialState,
      flowContext: {
        ...initialState.flowContext,
        definition: boring,
        nodes: testNodes,
      },
    },
  };
};

export const createMockStore: Function = configureStore([thunk]);

export const mock = <T extends {}, K extends keyof T>(
  object: T,
  property: K,
  value: T[K],
) => {
  Object.defineProperty(object, property, { get: () => value });
};
