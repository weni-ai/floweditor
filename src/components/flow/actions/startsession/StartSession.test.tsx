import StartSessionComp from 'components/flow/actions/startsession/StartSession';
import { StartSession } from 'flowTypes';
import React from 'react';
import { render } from 'test/utils';
import { createStartSessionAction } from 'testUtils/assetCreators';

describe('StartSessionComp', () => {
  const baseProps: StartSession = createStartSessionAction();

  describe('render', () => {
    it('should render all props', () => {
      const { baseElement } = render(<StartSessionComp {...baseProps} />);
      expect(baseElement).toMatchSnapshot();
    });

    it('should render a long list of both contacts and proups', () => {
      const { baseElement, getByText } = render(
        <StartSessionComp
          {...baseProps}
          contacts={[
            { uuid: 'contact-1', name: 'Norbert Kwizera' },
            { uuid: 'contact-2', name: 'Kellan Alexander' },
            { uuid: 'contact-3', name: 'Rowan Seymour' },
            { uuid: 'contact-4', name: 'Leah Burgerbuns' },
            { uuid: 'contact-5', name: 'Nic Pottier' },
          ]}
        />,
      );

      expect(baseElement).toMatchSnapshot();
      getByText('+2 more');
    });

    it('should render creating a new contact', () => {
      const { baseElement, getByText } = render(
        <StartSessionComp
          {...baseProps}
          contacts={[]}
          groups={[]}
          create_contact={true}
        />,
      );
      expect(baseElement).toMatchSnapshot();
      getByText('Create a new contact');
    });
  });
});
