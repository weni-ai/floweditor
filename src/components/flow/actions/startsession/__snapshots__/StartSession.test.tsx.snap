// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StartSessionComp > render > should render a long list of both contacts and proups 1`] = `
<body>
  <div>
    <div
      class="_node_9ae69d"
    >
      <div
        class="_to_9ae69d"
      >
        <div
          class="_node_asset_9658e1"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="display: inline-block;"
          >
            <svg
              class="unnnic-icon unnnic-icon__size--avatar-nano unnnic-icon-scheme--neutral-dark _icon_25b42d"
              data-v-930380cc=""
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="primary"
                d="M20 20.0005C15.865 20.0005 12.5 16.6355 12.5 12.5005C12.5 8.36549 15.865 5.00049 20 5.00049C24.135 5.00049 27.5 8.36549 27.5 12.5005C27.5 16.6355 24.135 20.0005 20 20.0005ZM20 6.87549C16.8988 6.87549 14.375 9.39924 14.375 12.5005C14.375 15.6017 16.8988 18.1255 20 18.1255C23.1012 18.1255 25.625 15.6017 25.625 12.5005C25.625 9.39924 23.1012 6.87549 20 6.87549Z"
                fill="#3B414D"
              />
              <path
                class="primary"
                d="M32.1875 35.0005C31.9389 35.0005 31.7004 34.9017 31.5246 34.7259C31.3488 34.5501 31.25 34.3116 31.25 34.063C31.25 27.8592 26.2038 22.813 20 22.813C13.7963 22.813 8.75 27.8592 8.75 34.063C8.75 34.3116 8.65123 34.5501 8.47541 34.7259C8.2996 34.9017 8.06114 35.0005 7.8125 35.0005C7.56386 35.0005 7.3254 34.9017 7.14959 34.7259C6.97377 34.5501 6.875 34.3116 6.875 34.063C6.875 26.8255 12.7625 20.938 20 20.938C27.2375 20.938 33.125 26.8255 33.125 34.063C33.125 34.3116 33.0262 34.5501 32.8504 34.7259C32.6746 34.9017 32.4361 35.0005 32.1875 35.0005Z"
                fill="#3B414D"
              />
            </svg>
          </div>
          <span
            class="_link_25b42d _truncated_25b42d"
          >
            Cat Fanciers
          </span>
        </div>
        <div
          class="_node_asset_9658e1"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="display: inline-block;"
          >
            <svg
              class="unnnic-icon unnnic-icon__size--avatar-nano unnnic-icon-scheme--neutral-dark _icon_25b42d"
              data-v-930380cc=""
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="primary"
                d="M20 20.0005C15.865 20.0005 12.5 16.6355 12.5 12.5005C12.5 8.36549 15.865 5.00049 20 5.00049C24.135 5.00049 27.5 8.36549 27.5 12.5005C27.5 16.6355 24.135 20.0005 20 20.0005ZM20 6.87549C16.8988 6.87549 14.375 9.39924 14.375 12.5005C14.375 15.6017 16.8988 18.1255 20 18.1255C23.1012 18.1255 25.625 15.6017 25.625 12.5005C25.625 9.39924 23.1012 6.87549 20 6.87549Z"
                fill="#3B414D"
              />
              <path
                class="primary"
                d="M32.1875 35.0005C31.9389 35.0005 31.7004 34.9017 31.5246 34.7259C31.3488 34.5501 31.25 34.3116 31.25 34.063C31.25 27.8592 26.2038 22.813 20 22.813C13.7963 22.813 8.75 27.8592 8.75 34.063C8.75 34.3116 8.65123 34.5501 8.47541 34.7259C8.2996 34.9017 8.06114 35.0005 7.8125 35.0005C7.56386 35.0005 7.3254 34.9017 7.14959 34.7259C6.97377 34.5501 6.875 34.3116 6.875 34.063C6.875 26.8255 12.7625 20.938 20 20.938C27.2375 20.938 33.125 26.8255 33.125 34.063C33.125 34.3116 33.0262 34.5501 32.8504 34.7259C32.6746 34.9017 32.4361 35.0005 32.1875 35.0005Z"
                fill="#3B414D"
              />
            </svg>
          </div>
          <span
            class="_link_25b42d _truncated_25b42d"
          >
            Norbert Kwizera
          </span>
        </div>
        <div
          class="_node_asset_9658e1"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="display: inline-block;"
          >
            <svg
              class="unnnic-icon unnnic-icon__size--avatar-nano unnnic-icon-scheme--neutral-dark _icon_25b42d"
              data-v-930380cc=""
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="primary"
                d="M20 20.0005C15.865 20.0005 12.5 16.6355 12.5 12.5005C12.5 8.36549 15.865 5.00049 20 5.00049C24.135 5.00049 27.5 8.36549 27.5 12.5005C27.5 16.6355 24.135 20.0005 20 20.0005ZM20 6.87549C16.8988 6.87549 14.375 9.39924 14.375 12.5005C14.375 15.6017 16.8988 18.1255 20 18.1255C23.1012 18.1255 25.625 15.6017 25.625 12.5005C25.625 9.39924 23.1012 6.87549 20 6.87549Z"
                fill="#3B414D"
              />
              <path
                class="primary"
                d="M32.1875 35.0005C31.9389 35.0005 31.7004 34.9017 31.5246 34.7259C31.3488 34.5501 31.25 34.3116 31.25 34.063C31.25 27.8592 26.2038 22.813 20 22.813C13.7963 22.813 8.75 27.8592 8.75 34.063C8.75 34.3116 8.65123 34.5501 8.47541 34.7259C8.2996 34.9017 8.06114 35.0005 7.8125 35.0005C7.56386 35.0005 7.3254 34.9017 7.14959 34.7259C6.97377 34.5501 6.875 34.3116 6.875 34.063C6.875 26.8255 12.7625 20.938 20 20.938C27.2375 20.938 33.125 26.8255 33.125 34.063C33.125 34.3116 33.0262 34.5501 32.8504 34.7259C32.6746 34.9017 32.4361 35.0005 32.1875 35.0005Z"
                fill="#3B414D"
              />
            </svg>
          </div>
          <span
            class="_link_25b42d _truncated_25b42d"
          >
            Kellan Alexander
          </span>
        </div>
        <div
          class="_node_asset_9658e1"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="display: inline-block;"
          >
            <svg
              class="unnnic-icon unnnic-icon__size--avatar-nano unnnic-icon-scheme--neutral-dark _icon_25b42d"
              data-v-930380cc=""
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="primary"
                d="M20 20.0005C15.865 20.0005 12.5 16.6355 12.5 12.5005C12.5 8.36549 15.865 5.00049 20 5.00049C24.135 5.00049 27.5 8.36549 27.5 12.5005C27.5 16.6355 24.135 20.0005 20 20.0005ZM20 6.87549C16.8988 6.87549 14.375 9.39924 14.375 12.5005C14.375 15.6017 16.8988 18.1255 20 18.1255C23.1012 18.1255 25.625 15.6017 25.625 12.5005C25.625 9.39924 23.1012 6.87549 20 6.87549Z"
                fill="#3B414D"
              />
              <path
                class="primary"
                d="M32.1875 35.0005C31.9389 35.0005 31.7004 34.9017 31.5246 34.7259C31.3488 34.5501 31.25 34.3116 31.25 34.063C31.25 27.8592 26.2038 22.813 20 22.813C13.7963 22.813 8.75 27.8592 8.75 34.063C8.75 34.3116 8.65123 34.5501 8.47541 34.7259C8.2996 34.9017 8.06114 35.0005 7.8125 35.0005C7.56386 35.0005 7.3254 34.9017 7.14959 34.7259C6.97377 34.5501 6.875 34.3116 6.875 34.063C6.875 26.8255 12.7625 20.938 20 20.938C27.2375 20.938 33.125 26.8255 33.125 34.063C33.125 34.3116 33.0262 34.5501 32.8504 34.7259C32.6746 34.9017 32.4361 35.0005 32.1875 35.0005Z"
                fill="#3B414D"
              />
            </svg>
          </div>
          <span
            class="_link_25b42d _truncated_25b42d"
          >
            Rowan Seymour
          </span>
        </div>
        <div>
          +
          2
           more
        </div>
      </div>
      <div
        class="_flow_9ae69d"
      >
        <div
          class="_node_asset_9658e1"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="display: inline-block;"
          >
            <svg
              class="unnnic-icon unnnic-icon__size--avatar-nano unnnic-icon-scheme--neutral-dark _icon_25b42d"
              data-v-930380cc=""
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="primary"
                d="M6.875 35.0005C5.84125 35.0005 5 34.1592 5 33.1255V29.3755C5 28.3417 5.84125 27.5005 6.875 27.5005H21.875C22.9088 27.5005 23.75 28.3417 23.75 29.3755V31.2505H26.5625C27.08 31.2505 27.5 30.8305 27.5 30.313V25.6255H18.125C17.0913 25.6255 16.25 24.7842 16.25 23.7505V20.0005C16.25 18.9667 17.0913 18.1255 18.125 18.1255H21.875V17.188C21.875 16.6705 21.455 16.2505 20.9375 16.2505H15.3125C13.7612 16.2505 12.5 14.9892 12.5 13.438V12.5005H6.875C5.84125 12.5005 5 11.6592 5 10.6255V6.87549C5 5.84174 5.84125 5.00049 6.875 5.00049H21.875C22.9088 5.00049 23.75 5.84174 23.75 6.87549V10.6255C23.75 11.6592 22.9088 12.5005 21.875 12.5005H14.375V13.438C14.375 13.9555 14.795 14.3755 15.3125 14.3755H20.9375C22.4888 14.3755 23.75 15.6367 23.75 17.188V18.1255H33.125C34.1588 18.1255 35 18.9667 35 20.0005V23.7505C35 24.7842 34.1588 25.6255 33.125 25.6255H29.375V30.313C29.375 31.8642 28.1138 33.1255 26.5625 33.1255H23.75C23.75 34.1592 22.9088 35.0005 21.875 35.0005H6.875ZM6.875 33.1255H21.875V29.3755H6.875V33.1255ZM18.125 23.7505H33.125V20.0005H18.125V23.7505ZM6.875 10.6255H21.875V6.87549H6.875V10.6255Z"
                fill="#3B414D"
              />
            </svg>
          </div>
          <a
            class="_link_25b42d"
            href="/flow/editor/flow_uuid"
            rel="noopener noreferrer"
            target="_blank"
          >
            Flow to Start
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StartSessionComp > render > should render all props 1`] = `
<body>
  <div>
    <div
      class="_node_9ae69d"
    >
      <div
        class="_to_9ae69d"
      >
        <div
          class="_node_asset_9658e1"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="display: inline-block;"
          >
            <svg
              class="unnnic-icon unnnic-icon__size--avatar-nano unnnic-icon-scheme--neutral-dark _icon_25b42d"
              data-v-930380cc=""
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="primary"
                d="M20 20.0005C15.865 20.0005 12.5 16.6355 12.5 12.5005C12.5 8.36549 15.865 5.00049 20 5.00049C24.135 5.00049 27.5 8.36549 27.5 12.5005C27.5 16.6355 24.135 20.0005 20 20.0005ZM20 6.87549C16.8988 6.87549 14.375 9.39924 14.375 12.5005C14.375 15.6017 16.8988 18.1255 20 18.1255C23.1012 18.1255 25.625 15.6017 25.625 12.5005C25.625 9.39924 23.1012 6.87549 20 6.87549Z"
                fill="#3B414D"
              />
              <path
                class="primary"
                d="M32.1875 35.0005C31.9389 35.0005 31.7004 34.9017 31.5246 34.7259C31.3488 34.5501 31.25 34.3116 31.25 34.063C31.25 27.8592 26.2038 22.813 20 22.813C13.7963 22.813 8.75 27.8592 8.75 34.063C8.75 34.3116 8.65123 34.5501 8.47541 34.7259C8.2996 34.9017 8.06114 35.0005 7.8125 35.0005C7.56386 35.0005 7.3254 34.9017 7.14959 34.7259C6.97377 34.5501 6.875 34.3116 6.875 34.063C6.875 26.8255 12.7625 20.938 20 20.938C27.2375 20.938 33.125 26.8255 33.125 34.063C33.125 34.3116 33.0262 34.5501 32.8504 34.7259C32.6746 34.9017 32.4361 35.0005 32.1875 35.0005Z"
                fill="#3B414D"
              />
            </svg>
          </div>
          <span
            class="_link_25b42d _truncated_25b42d"
          >
            Cat Fanciers
          </span>
        </div>
        <div
          class="_node_asset_9658e1"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="display: inline-block;"
          >
            <svg
              class="unnnic-icon unnnic-icon__size--avatar-nano unnnic-icon-scheme--neutral-dark _icon_25b42d"
              data-v-930380cc=""
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="primary"
                d="M20 20.0005C15.865 20.0005 12.5 16.6355 12.5 12.5005C12.5 8.36549 15.865 5.00049 20 5.00049C24.135 5.00049 27.5 8.36549 27.5 12.5005C27.5 16.6355 24.135 20.0005 20 20.0005ZM20 6.87549C16.8988 6.87549 14.375 9.39924 14.375 12.5005C14.375 15.6017 16.8988 18.1255 20 18.1255C23.1012 18.1255 25.625 15.6017 25.625 12.5005C25.625 9.39924 23.1012 6.87549 20 6.87549Z"
                fill="#3B414D"
              />
              <path
                class="primary"
                d="M32.1875 35.0005C31.9389 35.0005 31.7004 34.9017 31.5246 34.7259C31.3488 34.5501 31.25 34.3116 31.25 34.063C31.25 27.8592 26.2038 22.813 20 22.813C13.7963 22.813 8.75 27.8592 8.75 34.063C8.75 34.3116 8.65123 34.5501 8.47541 34.7259C8.2996 34.9017 8.06114 35.0005 7.8125 35.0005C7.56386 35.0005 7.3254 34.9017 7.14959 34.7259C6.97377 34.5501 6.875 34.3116 6.875 34.063C6.875 26.8255 12.7625 20.938 20 20.938C27.2375 20.938 33.125 26.8255 33.125 34.063C33.125 34.3116 33.0262 34.5501 32.8504 34.7259C32.6746 34.9017 32.4361 35.0005 32.1875 35.0005Z"
                fill="#3B414D"
              />
            </svg>
          </div>
          <span
            class="_link_25b42d _truncated_25b42d"
          >
            Norbert Kwizera
          </span>
        </div>
        <div
          class="_node_asset_9658e1"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="display: inline-block;"
          >
            <svg
              class="unnnic-icon unnnic-icon__size--avatar-nano unnnic-icon-scheme--neutral-dark _icon_25b42d"
              data-v-930380cc=""
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="primary"
                d="M20 20.0005C15.865 20.0005 12.5 16.6355 12.5 12.5005C12.5 8.36549 15.865 5.00049 20 5.00049C24.135 5.00049 27.5 8.36549 27.5 12.5005C27.5 16.6355 24.135 20.0005 20 20.0005ZM20 6.87549C16.8988 6.87549 14.375 9.39924 14.375 12.5005C14.375 15.6017 16.8988 18.1255 20 18.1255C23.1012 18.1255 25.625 15.6017 25.625 12.5005C25.625 9.39924 23.1012 6.87549 20 6.87549Z"
                fill="#3B414D"
              />
              <path
                class="primary"
                d="M32.1875 35.0005C31.9389 35.0005 31.7004 34.9017 31.5246 34.7259C31.3488 34.5501 31.25 34.3116 31.25 34.063C31.25 27.8592 26.2038 22.813 20 22.813C13.7963 22.813 8.75 27.8592 8.75 34.063C8.75 34.3116 8.65123 34.5501 8.47541 34.7259C8.2996 34.9017 8.06114 35.0005 7.8125 35.0005C7.56386 35.0005 7.3254 34.9017 7.14959 34.7259C6.97377 34.5501 6.875 34.3116 6.875 34.063C6.875 26.8255 12.7625 20.938 20 20.938C27.2375 20.938 33.125 26.8255 33.125 34.063C33.125 34.3116 33.0262 34.5501 32.8504 34.7259C32.6746 34.9017 32.4361 35.0005 32.1875 35.0005Z"
                fill="#3B414D"
              />
            </svg>
          </div>
          <span
            class="_link_25b42d _truncated_25b42d"
          >
            Rowan Seymour
          </span>
        </div>
      </div>
      <div
        class="_flow_9ae69d"
      >
        <div
          class="_node_asset_9658e1"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="display: inline-block;"
          >
            <svg
              class="unnnic-icon unnnic-icon__size--avatar-nano unnnic-icon-scheme--neutral-dark _icon_25b42d"
              data-v-930380cc=""
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="primary"
                d="M6.875 35.0005C5.84125 35.0005 5 34.1592 5 33.1255V29.3755C5 28.3417 5.84125 27.5005 6.875 27.5005H21.875C22.9088 27.5005 23.75 28.3417 23.75 29.3755V31.2505H26.5625C27.08 31.2505 27.5 30.8305 27.5 30.313V25.6255H18.125C17.0913 25.6255 16.25 24.7842 16.25 23.7505V20.0005C16.25 18.9667 17.0913 18.1255 18.125 18.1255H21.875V17.188C21.875 16.6705 21.455 16.2505 20.9375 16.2505H15.3125C13.7612 16.2505 12.5 14.9892 12.5 13.438V12.5005H6.875C5.84125 12.5005 5 11.6592 5 10.6255V6.87549C5 5.84174 5.84125 5.00049 6.875 5.00049H21.875C22.9088 5.00049 23.75 5.84174 23.75 6.87549V10.6255C23.75 11.6592 22.9088 12.5005 21.875 12.5005H14.375V13.438C14.375 13.9555 14.795 14.3755 15.3125 14.3755H20.9375C22.4888 14.3755 23.75 15.6367 23.75 17.188V18.1255H33.125C34.1588 18.1255 35 18.9667 35 20.0005V23.7505C35 24.7842 34.1588 25.6255 33.125 25.6255H29.375V30.313C29.375 31.8642 28.1138 33.1255 26.5625 33.1255H23.75C23.75 34.1592 22.9088 35.0005 21.875 35.0005H6.875ZM6.875 33.1255H21.875V29.3755H6.875V33.1255ZM18.125 23.7505H33.125V20.0005H18.125V23.7505ZM6.875 10.6255H21.875V6.87549H6.875V10.6255Z"
                fill="#3B414D"
              />
            </svg>
          </div>
          <a
            class="_link_25b42d"
            href="/flow/editor/flow_uuid"
            rel="noopener noreferrer"
            target="_blank"
          >
            Flow to Start
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`StartSessionComp > render > should render creating a new contact 1`] = `
<body>
  <div>
    <div
      class="_node_9ae69d"
    >
      <div
        class="_to_9ae69d"
      >
        Create a new contact
      </div>
      <div
        class="_flow_9ae69d"
      >
        <div
          class="_node_asset_9658e1"
        >
          <div
            data-use-vue-component-wrap=""
            data-v-app=""
            style="display: inline-block;"
          >
            <svg
              class="unnnic-icon unnnic-icon__size--avatar-nano unnnic-icon-scheme--neutral-dark _icon_25b42d"
              data-v-930380cc=""
              fill="none"
              height="40"
              viewBox="0 0 40 40"
              width="40"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="primary"
                d="M6.875 35.0005C5.84125 35.0005 5 34.1592 5 33.1255V29.3755C5 28.3417 5.84125 27.5005 6.875 27.5005H21.875C22.9088 27.5005 23.75 28.3417 23.75 29.3755V31.2505H26.5625C27.08 31.2505 27.5 30.8305 27.5 30.313V25.6255H18.125C17.0913 25.6255 16.25 24.7842 16.25 23.7505V20.0005C16.25 18.9667 17.0913 18.1255 18.125 18.1255H21.875V17.188C21.875 16.6705 21.455 16.2505 20.9375 16.2505H15.3125C13.7612 16.2505 12.5 14.9892 12.5 13.438V12.5005H6.875C5.84125 12.5005 5 11.6592 5 10.6255V6.87549C5 5.84174 5.84125 5.00049 6.875 5.00049H21.875C22.9088 5.00049 23.75 5.84174 23.75 6.87549V10.6255C23.75 11.6592 22.9088 12.5005 21.875 12.5005H14.375V13.438C14.375 13.9555 14.795 14.3755 15.3125 14.3755H20.9375C22.4888 14.3755 23.75 15.6367 23.75 17.188V18.1255H33.125C34.1588 18.1255 35 18.9667 35 20.0005V23.7505C35 24.7842 34.1588 25.6255 33.125 25.6255H29.375V30.313C29.375 31.8642 28.1138 33.1255 26.5625 33.1255H23.75C23.75 34.1592 22.9088 35.0005 21.875 35.0005H6.875ZM6.875 33.1255H21.875V29.3755H6.875V33.1255ZM18.125 23.7505H33.125V20.0005H18.125V23.7505ZM6.875 10.6255H21.875V6.87549H6.875V10.6255Z"
                fill="#3B414D"
              />
            </svg>
          </div>
          <a
            class="_link_25b42d"
            href="/flow/editor/flow_uuid"
            rel="noopener noreferrer"
            target="_blank"
          >
            Flow to Start
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
